apps/meteor/tests/unit/server/startup/initialData.tests.ts

```diff
@@ -0,0 +1,235 @@
+import { expect } from 'chai';
+import { beforeEach, it } from 'mocha';
+import proxyquire from 'proxyquire';
+import sinon from 'sinon';
+
+const getUsersInRole = sinon.stub();
+const checkUsernameAvailability = sinon.stub();
+const validateEmail = sinon.stub();
+const addUserRolesAsync = sinon.stub();
+const models = {
+	Settings: {},
+	Rooms: {},
+	Users: {
+		create: sinon.stub(),
+		findOneByEmailAddress: sinon.stub(),
+	},
+};
+const setPasswordAsync = sinon.stub();
+const settingsGet = sinon.stub();
+
+const { insertAdminUserFromEnv } = proxyquire.noCallThru().load('../../../../server/startup/initialData.js', {
+	'meteor/accounts-base': {
+		Accounts: {
+			setPasswordAsync,
+		},
+	},
+	'meteor/meteor': {
+		Meteor: {
+			startup: sinon.stub(),
+		},
+	},
+	'../../app/authorization/server': {
+		getUsersInRole,
+	},
+	'../../app/file-upload/server': {},
+	'../../app/file/server': {},
+	'../../app/lib/server/functions/addUserToDefaultChannels': {},
+	'../../app/lib/server/functions/checkUsernameAvailability': {
+		checkUsernameAvailability,
+	},
+	'../../app/settings/server': {
+		settings: { get: settingsGet },
+	},
+	'../../lib/emailValidator': {
+		validateEmail,
+	},
+	'../lib/roles/addUserRoles': {
+		addUserRolesAsync,
+	},
+	'@rocket.chat/models': models,
+});
+
+describe('insertAdminUserFromEnv', () => {
+	beforeEach(() => {
+		getUsersInRole.reset();
+		checkUsernameAvailability.reset();
+		validateEmail.reset();
+		addUserRolesAsync.reset();
+		models.Users.create.reset();
+		models.Users.findOneByEmailAddress.reset();
+		setPasswordAsync.reset();
+		settingsGet.reset();
+		process.env.ADMIN_PASS = 'pass';
+	});
+
+	it('should do nothing if process.env.ADMIN_PASS is empty', async () => {
+		process.env.ADMIN_PASS = '';
+		const result = await insertAdminUserFromEnv();
+		expect(getUsersInRole.called).to.be.false;
+		expect(result).to.be.undefined;
+	});
+	it('should do nothing if theres already an admin user', async () => {
+		getUsersInRole.returns({ count: () => 1 });
+
+		const result = await insertAdminUserFromEnv();
+		expect(getUsersInRole.called).to.be.true;
+		expect(validateEmail.called).to.be.false;
+		expect(result).to.be.undefined;
+	});
+	it('should try to validate an email when process.env.ADMIN_EMAIL is set', async () => {
+		process.env.ADMIN_EMAIL = 'email';
+		getUsersInRole.returns({ count: () => 0 });
+		validateEmail.returns(false);
+		models.Users.create.returns({ insertedId: 'newuserid' });
+
+		const result = await insertAdminUserFromEnv();
+		expect(getUsersInRole.called).to.be.true;
+		expect(validateEmail.called).to.be.true;
+		expect(validateEmail.calledWith('email')).to.be.true;
+		expect(models.Users.create.called).to.be.true;
```

Comment:
Let's check if this was called with the proper email address.

Reply:
That's the test on L129. This one only checks that when ADMIN_EMAIL is set to any value, it will try to validate it.

L129 covers the email being valid and saved by users.create